### PR TITLE
Feature/UNR-2589 latency tracking

### PR DESCRIPTION
**Pull request by ImprobableVlad**
#### Context
This PR is part of an effort to produce, aggregate, upload and display NFR latency metrics.
The following 4 PR's constitute the required code repo changes for this goal:
* Nfr-Benchmark-Pipeline: https://github.com/improbable/nfr-benchmark-pipeline/pull/88
* UnrealGDKTestGyms: https://github.com/spatialos/UnrealGDKTestGyms/pull/23
* UnrealGDK: https://github.com/spatialos/UnrealGDK/pull/1717
* ShooterGameBuildkite: https://github.com/improbable/ShooterGameBuildKite/pull/22

#### Description of Changes
These changes make sure that the whole project is cooked, so that all maps are accessible. Furthermore the deployment name is now passed to the unreal worker, so that it can be used for extracting the map name if specified (which is done by the code introduced in the ShooterGameBuildkite PR). Additional logic has been added to the latency map blueprint to set trace prefixes when required. Finally, I have included an upgrade to `Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp` by @improbable-danny that fixes a null-pointer related error. Pictures of all changes to binary assets have been attached below.

#### Release note
These changes are not public facing, so no release note has been added.

#### Tests
A build that makes use of all four changed repositories has been kicked off at https://buildkite.com/improbable/shootergamebuildkite-sg-perf-test/builds/675. All NFR scenarios complete succesfully.

#### Documentation
In-code commentary has been provided.
Full documentation for the new latency tracing scenario has been provided in the Nfr Runner PR.

#### Primary reviewers
@improbable-danny  @m-samiec  @samiwh @michelefabris 

#### Pictures of Changes
latency_game_mode
latency_map
player_start

### CODE REVIEW

File: Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp [hunk 1]
```diff
@@ -84,42 +84,37 @@ bool ABenchmarkGymGameMode::ShouldUseCustomSpawning()
 void ABenchmarkGymGameMode::ParsePassedValues()
 {
 	USpatialNetDriver* NetDriver = Cast<USpatialNetDriver>(GetNetDriver());
```
Comment (improbable-danny) [inline Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp]:
```suggestion
	USpatialNetDriver* NetDriver = Cast<USpatialNetDriver>(GetNetDriver());
	
```

**Comment by ImprobableVlad** [comment]
After discussing this offline, I am removing `Game/Content/Characters/LatencyPlayerCharacter_BP.uasset` from this PR, as a recent commit by @improbable-danny has applied my changes to this asset already.

File: Game/Config/DefaultGame.ini [hunk 2]
```diff
@@ -5,3 +5,5 @@ bSpatialNetworking=True
 
 [/Script/UnrealEd.ProjectPackagingSettings]
 +DirectoriesToAlwaysCook=(Path="Spatial")
+bCookAll=True
```
Comment (m-samiec) [inline Game/Config/DefaultGame.ini]: Isn't this the default when not specifying maps? Can we just remove the `map` from the cook cmd line?

File: spatial/workers/unreal/spatialos.UnrealWorker.worker.json [hunk 3]
```diff
@@ -66,6 +66,8 @@
       "arguments": [
         "${IMPROBABLE_WORKER_ID}",
         "${IMPROBABLE_LOG_FILE}",
+        "+deploymentName",
```
Comment (m-samiec) [inline spatial/workers/unreal/spatialos.UnrealWorker.worker.json]: What's the reasoning behind this change? Should we change our default worker templates also?

File: Game/Config/DefaultGame.ini (hunk 2 above)
Comment (ImprobableVlad) [reply Game/Config/DefaultGame.ini]: Unfortunately it doesn't seem to be that way. When building by default (prior to this change) only the `EmptyGym` was packaged. I'm assuming it's controlled by either `Game Default Map` or `Server Default Map`: 
image

File: spatial/workers/unreal/spatialos.UnrealWorker.worker.json (hunk 3 above)
Comment (ImprobableVlad) [reply spatial/workers/unreal/spatialos.UnrealWorker.worker.json]: My custom `StartWorker.sh` that overrides the original in the assemlby uses this `deploymentName` parameter to extract any embedded map specification (for deployment names of the format `ANYTHING_map_MAPNAME_ANYTHINGELSE`). This `StartWorker.sh` override is only present in the `ShooterGameBuildkite` repo, so passing this extra parameter would not have any effect in other templates.

File: Game/Config/DefaultGame.ini (hunk 2 above)
Comment (m-samiec) [reply Game/Config/DefaultGame.ini]: Where's the call for cooking/packaging the game coming from?

File: spatial/workers/unreal/spatialos.UnrealWorker.worker.json (hunk 3 above)
Comment (m-samiec) [reply spatial/workers/unreal/spatialos.UnrealWorker.worker.json]: Cool - thanks for the info

File: Game/Config/DefaultGame.ini (hunk 2 above)
Comment (ImprobableVlad) [reply Game/Config/DefaultGame.ini]: [This line](https://github.com/improbable/ShooterGameBuildKite/blob/9fe4275e2a10bc9a2a82e0d15fc2d62ecccb627c/ci/setup-and-build.ps1#L178) builds the server and [This line](https://github.com/improbable/ShooterGameBuildKite/blob/9fe4275e2a10bc9a2a82e0d15fc2d62ecccb627c/ci/setup-and-build.ps1#L194) in the same file builds the simplayer coordinator.

Comment (m-samiec) [reply Game/Config/DefaultGame.ini]: Okay I follow

**Pull request merged**